src/Storage/Repository/PgInstanceRepository.cs

```diff
@@ -68,137 +83,177 @@ public async Task<bool> Delete(Instance item)
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.Integer, size);
-            if (partyIdStringValues != default(StringValues))
+            return command.ToString()[..^2] + ");";
+        }
+
+        private static string ArrayVariableFromText(string[] arr)
+        {
+            StringBuilder value = new("'{");
+            foreach (string param in arr)
             {
-                if (partyIdStringValues.Count == 1)
-                {
-                    pgcom.Parameters.AddWithValue(NpgsqlDbType.Integer, partyId);
-                }
-                else if (partyIdStringValues.Count > 1)
-                {
-                    pgcom.Parameters.AddWithValue(NpgsqlDbType.Array | NpgsqlDbType.Integer, partyIdArray);
-                }
+                value.Append($"\"{param}\", ");
             }
-            else
+
+            return value.ToString()[..^2] + "}'";
+        }
+
+        private static string ArrayVariableFromJsonText(string[] arr)
+        {
+            StringBuilder value = new("ARRAY [");
+            foreach (string param in arr)
             {
-                pgcom.Parameters.AddWithValue(NpgsqlDbType.Integer, 0);
+                value.Append($"'{param}', ");
             }
 
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, taskPredicate != null ? queryParams["process.currentTask"].First() : DBNull.Value);
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, orgPredicate != null ? queryParams["org"].First() : DBNull.Value);
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, appIdPredicate != null ? queryParams["appId"].First() : DBNull.Value);
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.TimestampTz, lastChanged);
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.TimestampTz, created);
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, hardDeletedPredicate != null ? queryParams["status.isHardDeleted"].First() : DBNull.Value);
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, softDeletedPredicate != null ? queryParams["status.isSoftDeleted"].First() : DBNull.Value);
-            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, archivedPredicate != null ? queryParams["status.isArchived"].First() : DBNull.Value);
+            return value.ToString()[..^2] + "]::jsonb[]";
+        }
+
+        private static string ArrayVariableFromInteger(int[] arr)
+        {
+            StringBuilder value = new("'{");
+            foreach (int param in arr)
+            {
+                value.Append($"{param}, ");
+            }
+
+            return value.ToString()[..^2] + "}'";
+        }
+
+        private async Task<InstanceQueryResponse> GetInstancesInternal(
+            Dictionary<string, StringValues> queryParams,
+            string continuationToken,
+            int size)
+        {
+            Instance instance = new(); // make sonarcloud happy
+            long id = -1;
```

Comment:
Move variables closer to where they are being used.

Reply:
Done